core/views/browse.py

```diff
@@ -197,26 +196,6 @@ def page(request, lccn, date, edition, sequence, words=None):
         else:
             explanation = ""
 
-    # if no word highlights were requests, see if the user came
```

Comment:
This logic looks like it still needs to exist do do the highlighting from the referring page's search keywords but the URL it generates can be a direct link rather than hitting `chronam_page_words` for an extra redirect.

The pedantic way to do this would be something like this:

```python
scheme, netloc, path, query, fragment = urlsplit(request.build_absolute_uri())
url = urlunsplit((scheme, netloc, path, query, words))
```

In this case, it'd probably be reasonable do 

```python
url = '%s?%s#%s' % (urlresolvers.reverse('chronam_page_words', kwargs=path_parts), request.GET.urlencode(), words)
```